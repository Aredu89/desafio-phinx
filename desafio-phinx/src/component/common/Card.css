.card{
  border-radius: 3px;
}
.card .image{
  height: 380px;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  border-radius: 3px;
  box-shadow: inset 0px -40px 125px 2px rgba(0,0,0,0.3);
}
.card .image .favorito{
  position: relative;
  top: 12px;
  float: right;
  right: 12px;
  cursor: pointer;
}
.card .image .name{
  position: relative;
  top: 331px;
  left: 15px;
  font-weight: 600;
  color: #fff;
  cursor: pointer;
}