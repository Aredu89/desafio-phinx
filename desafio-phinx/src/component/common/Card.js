import React from 'react';
import './Card.css';

//Imagenes
import StarBorderWhite from '../../images/icons/star_outline_white.svg'

const Card = props => {
  const { character } = props
  return(
    <div className="card">
      <div 
        className="image"
        style={{
          backgroundImage: `url("${character.thumbnail.path}.${character.thumbnail.extension}")`
        }}
        >
        <img className="favorito" src={StarBorderWhite} alt="Favorito" />
        <div className="name">{character.name}</div>
      </div>   
    </div>
  )
}

export default Card;