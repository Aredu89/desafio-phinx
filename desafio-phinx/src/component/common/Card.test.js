import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card.js';

describe('Card', () => {
  it('Render',()=>{
    const props = {
      character:{
        id: 1
      }
    }
    const component = shallow(<Card {...props} />);
    expect(component).toMatchSnapshot();
  })
})