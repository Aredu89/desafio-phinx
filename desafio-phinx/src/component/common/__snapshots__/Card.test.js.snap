// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Render 1`] = `
<div
  className="card"
>
  <div
    className="image"
    style={
      Object {
        "backgroundImage": "url(\\".\\")",
      }
    }
  >
    <img
      alt="Favorito"
      className="favorito"
      onClick={[Function]}
      src="star_outline_white.svg"
    />
    <div
      className="name"
      onClick={[Function]}
    />
  </div>
</div>
`;
